app/controllers/common_attribute_states_controller.rb

```diff
@@ -28,10 +28,7 @@ def create
       @common_attribute_states.each do |cas|
         raise ActiveRecord::Rollback unless cas.save
       end
-      unless @room.update(last_time_checked: DateTime.now)
-        flash.now['alert'] = "Error updating room record"
-        return
-      end
+      raise ActiveRecord::Rollback unless @room.update(last_time_checked: DateTime.now)
     end
 
     if @common_attribute_states.all?(&:persisted?)
```

Comment:
if you want to pass to the next page _if the transaction failed_ instead, we can do this:

```
transaction = ActiveRecord::Base.transaction do
  @common_attribute_states.each do |cas|
    raise ActiveRecord::Rollback unless cas.save
  end
  raise ActiveRecord::Rollback unless @room.update(last_time_checked: DateTime.now)
  true
end

if transaction
  redirect_to redirect_rover_to_correct_state(room: @room, room_state: @room_state, step: "common_attributes", mode: "new")
else
  render :new, status: :unprocessable_entity
end
```